Xamarin.PropertyEditing.Mac/Controls/Custom/ColorEditorView.cs

```diff
@@ -43,6 +43,10 @@ public virtual void OnViewModelChanged (TViewModel oldModel)
 		public virtual void OnPropertyChanged (object sender, PropertyChangedEventArgs e)
 		{
 		}
+
+		protected virtual void AppearanceChanged ()
```

Comment:
Can we add this declaration here:

```
public sealed override void ViewDidChangeChangeEffectiveAppearance ()
{
    base.ViewDidChangeEffectiveAppearance ();
    AppearanceChanged ();
}
```
This gives us two main advantage:
1) We are guaranteed that ViewDidChangeEffectiveAppearance is overridden once for all subclasses, so AppearanceChanged will only be invoked once every time ViewDidChangeEffectiveAppearance is invoked. Without this, if there are multiple subclasses then each of them may override ViewDidChangeEffectiveAppearance and invoke AppearanceChanged, which means we'll get multiple invocations of AppearanceChanged.
2) It ensures that no-one accidentally puts code in ViewDidChangeEffectiveAppearance in other subclasses.

On a related note, we should probably seal every override of this method for the reasons :)

Reply:
@alanmcgovern When you say *here*, I assume you actually mean in the actual base class. Just asking to be 100% clear.

Reply:
Hmm Nm, This makes sense, but highlights I need to do this in the *other* base class.

Reply:
Done!

Reply:
Ah yeah, i meant in this base class and all other places where the method is overridden :)